flang/lib/Evaluate/intrinsics.cpp

```diff
@@ -1450,6 +1450,17 @@ static const IntrinsicInterface intrinsicSubroutine[]{
             {"errmsg", DefaultChar, Rank::scalar, Optionality::optional,
                 common::Intent::InOut}},
         {}, Rank::elemental, IntrinsicClass::collectiveSubroutine},
+    {"co_reduce",
+        {{"a", AnyData, Rank::anyOrAssumedRank, Optionality::required,
```

Comment:
why do you accept an assumed rank dummy argument?

Reply:
Hi, because, according to the standard, A can be a scalar or an array without specifying whether it can be assumed or not. And the definition of A is almost identical to those of `co_min`, `co_max` and so on, without the restriction for the type.

Reply:
If an assumed rank dummy argument were acceptable here, the standard would explicit say so.

See constraint C840 in subclause 8.5.8.7 in F'2023.

Reply:
Ok, in that case I agree. I'll update that.